esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,346 @@
+did not change the model code for this aim yet.
+"""
+import logging
+from pathlib import Path
+
+import numpy as np
+import xarray as xr
+import pandas as pd
+import dask.array as da
+import iris
+
+from esmvalcore.preprocessor import regrid
+from esmvaltool.diag_scripts.hydrology.derive_evspsblpot import debruin_pet
+from esmvaltool.diag_scripts.hydrology.compute_chunks import compute_chunks
+from esmvaltool.diag_scripts.shared import (ProvenanceLogger,
+                                            get_diagnostic_filename,
+                                            group_metadata,
+                                            run_diagnostic)
+
+
+logger = logging.getLogger(Path(__file__).name)
+
+
+def create_provenance_record():
+    """Create a provenance record."""
+    record = {
+        'caption': "Forcings for the GlobWat hydrological model.",
+        'domains': ['global'],
+        'authors': [
+            'abdollahi_banafsheh',
+            'alidoost_sarah',
+        ],
+        'projects': [
+            'ewatercycle',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': [],
+    }
+    return record
+
+
+def rechunk_and_regrid(src, tgt, scheme):
+    """Rechunk cube src and regrid it onto the grid of cube tgt."""
+    src_chunks = compute_chunks(src, tgt)
+    src.data = src.lazy_data().rechunk(src_chunks)
+    return regrid(src, tgt, scheme)
+
+
+def change_data_type(cube):
+    """Change data type to float32."""
+    cube.data = cube.core_data().astype('float32')
+    for coord_name in 'latitude', 'longitude', 'time':
+        coord = cube.coord(coord_name)
+        coord.points = coord.core_points().astype('float32')
+        coord.bounds = None
+        coord.guess_bounds()
+    return cube
+
+
+def _convert_units(cube):
+    """Convert unit of cube.
+
+    From kg m-2 s-1 to kg m-2 month-1 or kg m-2 day-1.
+    Note that the unit kg m-2 s-1 is equivalent to mm s-1.
+    """
+    mip = cube.attributes['mip']
+
+    if mip == 'Amon':
+        cube.convert_units('kg m-2 month-1')  # equivalent to mm/month
+    elif mip == 'day':
+        cube.convert_units('kg m-2 day-1')  # equivalent to mm/day
+    return cube
+
+
+def _fix_negative_values(cube):
+    """Change negative values to zero."""
+    cube.data = da.where(cube.core_data() < 0, 0, cube.core_data())
+    return cube
+
+
+def get_input_cubes(metadata):
+    """Return a dictionary with all (preprocessed) input files."""
+    provenance = create_provenance_record()
+    all_vars = {}
+    for attributes in metadata:
+        short_name = attributes['short_name']
+        filename = attributes['filename']
+        logger.info("Loading variable %s", short_name)
+        cube = iris.load_cube(filename)
+        all_vars[short_name] = change_data_type(cube)
+        cube.attributes['mip'] = attributes['mip']
+        provenance['ancestors'].append(filename)
+    return all_vars, provenance
+
+
+def load_target(cfg):
```

Comment:
Okay so the `target_file` is used to obtain the `target_grid`? I was a bit confused by this, as target file sounds to me like you're going to write the eventual output to this particular file. Perhaps consider renaming this to something like `target_grid_file`?

Reply:
I renamed it.